src/libstd/time/duration.rs

```diff
@@ -188,6 +188,13 @@ impl DivAssign<u32> for Duration {
     }
 }
 
+#[stable(feature = "duration_default", since = "1.9.0")]
+impl Default for Duration {
+    fn default() -> Duration {
+        Duration::new(0, 0)
```

Comment:
The problem is that Durations are often used as Timeouts and unless zero-duration timeouts are special case, the operation will *always* time out by default.